### PR TITLE
FIX: Improve Onebox detection

### PR DESCRIPTION
**Pull request by nbianca**
Follow-up to 7c83d2eeb261ac676a8320e6a704752c56fd242e.

### CODE REVIEW

**Comment by ZogStriP** [comment]
Looks good, but the build is broken. Can you have a look?

File: app/models/post_analyzer.rb [hunk 1]
```diff
@@ -36,7 +37,9 @@ def cook(raw, opts = {})
     result = Oneboxer.apply(cooked) do |url|
       @onebox_urls << url
```
Comment (ZogStriP) [inline app/models/post_analyzer.rb]: Should we keep track of `onebox_urls` if `Oneboxer.cached_onebox` might fail later?

File: app/models/post_analyzer.rb [hunk 2]
```diff
@@ -126,7 +129,7 @@ def raw_links
 
   # How many links are present in the post
   def link_count
-    raw_links.size
+    raw_links.size + @onebox_urls.size
```
Comment (ZogStriP) [inline app/models/post_analyzer.rb]: Why that change?

Comment (nbianca) [reply app/models/post_analyzer.rb]: Some links were either in the `raw_links` array or in the `@onebox_urls` array, so `link_count` was returning only a partial count of the links present in the post.

File: app/models/post_analyzer.rb (hunk 1 above)
Comment (nbianca) [reply app/models/post_analyzer.rb]: `@onebox_urls` keeps all links that should be Oneboxed whether they succeed or not. It is used later to determine the number of links.

**Pull request merged**